notebooks/gotcha.clj

```diff
@@ -0,0 +1,27 @@
+(ns gotcha
+  [:require
+   [scicloj.kindly.v4.kind :as k]])
+
+(k/md "# Gotchas...
+Although we try to avoid such things, sometimes, when you're fighting the host language, it's just not practical to carry over the original conventions. Here we will try to keep an up-to-date list of possible surprises (that are not easy to 'fix').
+
+**^** (`Power`) - Despite allowing a wider range of characters, we cannot use `^` in Wolframite because it is reserved for adding metadata to virtually all Clojure symbols. `(^ x 2)` will probably not do what you expect!
+
+**:=** (`SetDelayed`) - Is actually `_=` in Wolframite. This is because `:` is usually a reserved in Clojure for creating keywords. It's worth noting though that this is possibly nicer, in a way, because `_` looks like a placeholder, and defining a placeholder for the expression is what it does. It also sort of implies a delay...
+
+**:>** (`RuleDelayed`) - Is `_>` in Wolframite.
+
+**.** (`Dot`) - Has been changed to `<*>`, in the spirit of the inner product, because `.` is a key character in Clojure, and most object-oriented systems, for namespacing.
+
+**/.** (`ReplaceAll`) -  Similarly, has been changed to `x>>` (and `Replace` to `x>`) for consistency.
+
+**=.** (`Unset`) -  has also been changed to `=!`; be careful not to confuse this with `!=`!
+
+Symbols within threading macros. - After spending so much time playing with symbols, be careful of slipping into things like
+`(-> 'x '(Power 1))`
+This will not work because threads are macros and so combining them with unevaluated functions will lead to errors.
```

Comment:
Why is the fact that `->` is a macro essential here?

Also, I'd be clearer:

```suggestion
This will not work because threads are macros and so combining them with unevaluated functions (`'(Power ...)` in this case) will lead to errors.
```

> will lead to errors

Why? What kind of errors? How should I then rewrite the expression to avoid said errors?

Reply:
This might be a mistake on my part, maybe it also wouldn't work with functions. I guess the emphasis should be that (-> 'x '(Power 1)) will not work, but that '(Power x 1) will - and losing the ability to chain expressions is a big loss.